numpy/lib/npyio.py

```diff
@@ -599,13 +611,41 @@ def savez_compressed(file, *args, **kwds):
     """
     _savez(file, args, kwds, True)
 
+@contextlib.contextmanager
```

Comment:
@njsmith I believe that for *real* exception handlers `__exit__` is called, then the exception is reraised.